custodial-copy-backend/src/main/scala/uk/gov/nationalarchives/custodialcopy/OcflService.scala

```diff
@@ -115,10 +116,11 @@ class OcflService(ocflRepository: MutableOcflRepository, semaphore: Semaphore[IO
     semaphore.acquire >> IO.blocking(ocflRepository.purgeObject(objectId.toString)).onError(logErrorAndRelease) >> semaphore.release
   }
 
-  private def isChecksumUnchanged(fixitiesMap: Map[DigestAlgorithm, String], checksums: List[Checksum]): Boolean = {
-    if (fixitiesMap.size != checksums.size) false
-    else checksums.forall(eachChecksum => fixitiesMap.get(DigestAlgorithm.fromOcflName(eachChecksum.algorithm.toLowerCase)).contains(eachChecksum.fingerprint))
-  }
+  private def isChecksumUnchanged(fixitiesMap: Map[DigestAlgorithm, String], checksums: List[Checksum]): IO[Boolean] =
+    for
+      psChecksum <- IO.fromOption(checksums.find(_.algorithm.toLowerCase == "sha256").map(_.fingerprint))(new Exception(s"SHA256 checksum missing in PS"))
+      ocflChecksum <- IO.fromOption(fixitiesMap.get(DigestAlgorithmRegistry.sha256))(new Exception(s"SHA256 checksum missing in OCFL"))
```

Comment:
If a 256 checksum is missing from both, it will only throw the PS exception, right? Should we collect the 1st exception and throw it along with the OCFL exception here?

Reply:
I think it's unlikely but I've added in some `ValidatedNel` bits which accumulates the errors and I've added a test to check for it.